test/Dialect/Torch/canonicalize.mlir

```diff
@@ -965,6 +965,15 @@ func.func @torch.prim.If$fold_same_result$subset_of_results(%arg0: !torch.bool,
   return %0, %1: !torch.int, !torch.int
 }
 
+// CHECK-LABEL: func.func @prim.ListConstruct$fold_list(
+// CHECK-SAME:      %[[ARG0:.*]]: !torch.list<tensor>) -> !torch.list<tensor> {
+// CHECK:         return %[[ARG0]] : !torch.list<tensor>
+func.func @prim.ListConstruct$fold_list(%arg0: !torch.list<tensor>) -> !torch.list<tensor> {
+  %0:2 = torch.prim.ListUnpack %arg0 : !torch.list<tensor> -> !torch.tensor, !torch.tensor
+  %1 = torch.prim.ListConstruct %0#0, %0#1 : (!torch.tensor, !torch.tensor) -> !torch.list<tensor>
+  return %1 : !torch.list<tensor>
+}
```

Comment:
I think it's necessary to check that `%arg0`'s length is same as `NumResults` of `prim.Listunpack`.

Reply:
We only replace the later ListConstruct op with the former ListUnpack's operand. ListUnpack op will not be eliminated.